packages/@aws-cdk/cloudformation-include/test/nested-stacks.test.ts

```diff
@@ -0,0 +1,528 @@
+    const parentTemplate = new inc.CfnInclude(stack, 'ParentStack', {
+      templateFile: testTemplateFilePath('cross-stack-refs.json'),
+      nestedStacks: {
+        'ChildStack': {
+          templateFile: testTemplateFilePath('child-import-stack.json'),
+        },
+      },
+    });
+    const cfnBucket = parentTemplate.getResource('Bucket');
+    cfnBucket.overrideLogicalId('DifferentBucket');
+    const parameter = parentTemplate.getParameter('Param');
+    parameter.overrideLogicalId('DifferentParameter');
+
+    expect(stack).toHaveResourceLike('AWS::CloudFormation::Stack', {
+      "Parameters": {
+        "Param1": {
+          "Ref": "DifferentParameter",
+        },
+        "Param2": {
+          "Fn::GetAtt": ["DifferentBucket", "Arn"],
+        },
+      },
+    });
+  });
+
+  test('ensure that getResource() returns a CfnStack', () => {
+    const cfnTemplate = new inc.CfnInclude(stack, 'ParentStack', {
+      templateFile: testTemplateFilePath('parent-export-stack.json'),
+    });
+
+    const childStack1 = cfnTemplate.getResource('ChildStack');
+
+    expect(childStack1).toBeInstanceOf(core.CfnStack);
+  });
+
+  test('attributes are correctly parsed', () => {
+    new inc.CfnInclude(stack, 'ParentStack', {
+      templateFile: testTemplateFilePath('parent-with-attributes.json'),
+    });
+
+    expect(stack).toHaveResourceLike('AWS::CloudFormation::Stack', {
+      "Properties": {
+        "TemplateURL": "https://cfn-templates-set.s3.amazonaws.com/child-import-stack.json",
+      },
+      "Metadata": {
+        "Property1": "Value1",
+      },
+      "DeletionPolicy": "Retain",
+      "DependsOn": [
+        "ChildStack",
+      ],
+      "UpdateReplacePolicy": "Retain",
+    }, ResourcePart.CompleteDefinition);
+  });
+
+  test('correctly parses NotificationsARNs, Timeout', () => {
+    new inc.CfnInclude(stack, 'ParentStack', {
+      templateFile: testTemplateFilePath('parent-with-attributes.json'),
+    });
+
+    expect(stack).toHaveResourceLike('AWS::CloudFormation::Stack', {
+      "Type": "AWS::CloudFormation::Stack",
+      "Properties": {
+        "TemplateURL": "https://cfn-templates-set.s3.amazonaws.com/child-import-stack.json",
+        "NotificationARNs": ["arn1"],
+        "TimeoutInMinutes": 5,
+      },
+    }, ResourcePart.CompleteDefinition);
+  });
+
+  test('correctly modifies the templateUrl property of a nested stack whose template has been passed', () => {
+    new inc.CfnInclude(stack, 'ParentStack', {
+      templateFile: testTemplateFilePath('only-nested-stack.json'),
+      nestedStacks: {
+        'NestedStack': {
+          // doesn't matter in this test
+          templateFile: testTemplateFilePath('child-import-stack.json'),
+        },
+      },
+    });
+
+    expect(stack).toMatchTemplate(
+      loadTestFileToJsObject('only-nested-stack.expected.json'),
+    );
+  });
+});
+
+describe('asset parameters are correctly passed from parent to grandchild', () => {
```

Comment:
This should be under the other `describe` above.

Reply:
I'm sorry, I don't know what you mean here.

Reply:
```ts
describe('CDK Include', () => {
    test('can ingest a template with one child', () => {
      // ...
    });

    // ...

    describe('asset parameters are correctly passed from parent to grandchild', () => {
       // ...
    });
});
```